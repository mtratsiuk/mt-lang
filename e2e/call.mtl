(print (+ 1 -2))

(print
  (-
    1
    2))

(console/log)

(print (+ 2 2))
(print (= !true false))
(print (and true false))
(print (or true false))
(print (< 10 5))
(print (>= 10 5))
(print (<= 10 5))
