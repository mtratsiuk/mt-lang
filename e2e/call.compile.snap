const __mtl_curry = f => (...args) => {
  if (args.length >= f.length) {
    return f(...args);
  }

  return __mtl_curry(f.bind(null, ...args));
};

console.log(String((1 + -2)));

console.log(String((1 - 2)));

console.log();

console.log(String((2 + 2)));

console.log(String((!true === false)));

console.log(String((true && false)));

console.log(String((true || false)));

console.log(String((10 < 5)));

console.log(String((10 >= 5)));

console.log(String((10 <= 5)));
